### PR TITLE
Move series.remove_na to core.dtypes.missing.remove_na_arraylike

### PR DESCRIPTION
**Pull request by jbrockmendel**
See discussion on #16931.

### CODE REVIEW

**Comment by jreback** [comment]
ignore the pep8speaks for now
we r testing it out

File: pandas/core/dtypes/missing.py [hunk 1]
```diff
@@ -394,3 +394,10 @@ def na_value_for_dtype(dtype):
     elif is_bool_dtype(dtype):
         return False
     return np.nan
+
+
+def remove_na_arraylike(series):
+    """
+    Return series containing only true/non-NaN values, possibly empty.
+    """
```
Comment (gfyoung) [inline pandas/core/dtypes/missing.py]: I know you and @jreback discussed this briefly, but I'd like to understand this:

Why are you saying "array-like" when the function only accepts `Series` (or is only meant for `Series`)?  If it isn't meant just for `Series`, we should update the documentation to reflect this.

Comment (gfyoung) [inline pandas/core/dtypes/missing.py]: Let's add tests as well for this function just to make sure it behaves like we want it too.

**Comment by gfyoung** [comment]
@jbrockmendel : Just a minor style error, but otherwise, everything looks green. 👍

**Comment by jbrockmendel** [comment]
What's the style error?

**Comment by gfyoung** [comment]
The style error can be found <a href="https://travis-ci.org/pandas-dev/pandas/jobs/253806510#L1779">here</a>.

File: pandas/core/dtypes/missing.py (hunk 1 above)
Comment (jreback) [inline pandas/core/dtypes/missing.py]: ok with tests. the name is fine. its descriptive and actually works on arrays as well. 

yes if you can throw in a couple of tests (with parametrize) would be great.

Comment (gfyoung) [inline pandas/core/dtypes/missing.py]: In that case, we should rename the variable name to something like `array_like` along with a description of accepted data types.

Comment (jreback) [inline pandas/core/dtypes/missing.py]: sure you can name it arr (our typical name)

**Comment by jreback** [comment]
lgtm

ping on green

**Comment by gfyoung** [comment]
Thanks @jbrockmendel !

**Pull request merged**